ui/nuxeo-document-picker/nuxeo-document-picker.js

```diff
@@ -0,0 +1,231 @@
+        /**
+         * List of content enrichers passed on to `provider`.
+         */
+        enrichers: String,
+        /**
+         * The base url for the hrefs of the search layouts to be loaded.
+         */
+        hrefBase: String,
+        /**
+         * The number of results per page.
+         */
+        pageSize: Number,
+        /**
+         * The id of the `nuxeo-page-provider` instance used to perform the search.
+         */
+        provider: String,
+        /**
+         * List of comma separated values of the document schemas to be returned.
+         * All document schemas are returned by default.
+         */
+        schemas: String,
+        /**
+         * The name of the search layout.
+         */
+        searchName: String,
+        /**
+         * The aggregations returned by `provider`.
+         */
+        _aggregations: {
+          type: Object,
+          readOnly: true,
+        },
+        /**
+         * The query parameters passed on to `provider`.
+         */
+        _params: {
+          type: Object,
+          readOnly: true,
+        },
+        /**
+         * Name of the quick filters to be displayed in case you don't want to display all of them.
+         * Expected format : ['quickfilter1','quickfilter2'].
+         */
+        _quickFilters: {
+          type: Object,
+          readOnly: true,
+        },
+      };
+    }
+
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      if (this._listenedResults && this._boundUpdateFn) {
+        this._listenedResults.removeEventListener('selected-items-changed', this._boundUpdateFn);
+      }
+    }
+
+    open() {
+      if (this.$.resultsView) {
+        // XXX could make sense to consider having this as a public method on nuxeo-results-view
+        this.$.resultsView._clear();
+      }
+      this._updateSelectButton();
+      this.$.dialog.open();
+    }
+
+    close() {
+      this.$.dialog.close();
+    }
+
+    get _selectedItems() {
+      return (
+        this.$.resultsView &&
+        this.$.resultsView.results &&
+        this.$.resultsView.results.results &&
+        this.$.resultsView.results.results.selectedItems
+      );
+    }
+
+    _updateSelectButton() {
+      const selectedItems = this._selectedItems;
+      this.$.selectButton.disabled = !(selectedItems && selectedItems.length);
+    }
+
+    _onSelect() {
+      const selectedItems = this._selectedItems;
+      if (selectedItems) {
+        this.dispatchEvent(
+          new CustomEvent('picked', {
+            composed: true,
+            bubbles: true,
+            detail: { selectedItems },
+          }),
+        );
+        this.close();
+      }
+    }
+
+    _onNavigate(e) {
+      // a grid thumbnail was clicked, so we want to select that result
+      this.$.resultsView.results.results.selectItems([e.detail.item]);
+      // XXX check how we can achieve a similar result for tables and the justified grid (not the default use case)
```

Comment:
Did we do this check?

Reply:
No. So far we only control the navigate events sent from data grid.

Reply:
I assume it would work as long as the results layout fires an navigate event, like we do on [nuxeo-default-search-results.html](https://github.com/nuxeo/nuxeo-web-ui/blob/master/elements/search/default/nuxeo-default-search-results.html#L241-L243)?

Reply:
something like:
```
_navigate(e) {
  this.fire('navigate', { item: (e.model || e.detail).item });
},
```

Reply:
Absolutely. The results layout can fire this event as it will bubble through `<nuxeo-search-results-layout>`, `<nuxeo-results-view>` and end up here in the `<nuxeo-document-picker>`. That's pretty much the same route the event from the data grid takes.

Reply:
From my understanding, this should work out of the box with a `nuxeo-data-list`, using as template a `nuxeo-document-list-item`, which is the most usual case, [right](https://github.com/nuxeo/nuxeo-web-ui/blob/master/elements/nuxeo-data-list/nuxeo-document-list-item.js#L266)? If so, we are just missing the justified grid and the table. It seems to me that table would be working as well if we just added to the nuxeo-results-view a `on-row-clicked="_onNavigate"`, no?

Reply:
https://user-images.githubusercontent.com/4223240/113759976-9f6caa80-970d-11eb-8ead-f41d5e32ac04.mp4

`<nuxeo-data-list>` does work out of the box using the following template:
```
<nuxeo-data-list
  name="list"
  icon="nuxeo:view-headline"
  empty-label="[[emptyLabel]]"
  multi-selection
  empty-label-when-filtered="[[emptyLabelWhenFiltered]]"
  selection-enabled
>
  <template>
    <nuxeo-document-list-item
      selected$="{{selected}}"
      doc="[[item]]"
      index="[[index]]"
      on-navigate="_navigate"
      selected-items="[[selectedItems]]"
    >
    </nuxeo-document-list-item>
  </template>
</nuxeo-data-list>
```

Reply:
https://user-images.githubusercontent.com/4223240/113760324-fe322400-970d-11eb-9b10-6c9933f26f89.mp4

On the other hand, when using `<nuxeo-data-table>` with the following template:
```
<nuxeo-data-table 
  class="results"
  settings-enabled
  empty-label="[[i18n('searchResults.noResults')]]"
  selection-enabled
  on-row-clicked="_navigate"
  icon="nuxeo:view-list"
  name="table"
>
  <nuxeo-data-table-column name="Title" field="dc:title" sort-by="dc:title" flex="100">
    <template>
      <nuxeo-document-thumbnail document="[[item]]"></nuxeo-document-thumbnail>
      <span>[[item.title]]</span>
    </template>
  </nuxeo-data-table-column>
  <nuxeo-data-table-column name="Modified" field="dc:modified" sort-by="dc:modified" flex="50">
    <template>[[formatDate(item.properties.dc:modified)]]</template>
  </nuxeo-data-table-column>
  <nuxeo-data-table-column name="Last Contributor" field="dc:lastContributor" sort-by="dc:lastContributor" flex="50">
    <template>
      <nuxeo-user-tag user="[[item.properties.dc:lastContributor]]"></nuxeo-user-tag>
    </template>
  </nuxeo-data-table-column>
</nuxeo-data-table>
```
You do have to provide a `_navigate` method:
```
_navigate: function(e) {
  this.fire('navigate', { item: (e.model || e.detail).item });
}
```
But even then the behaviour is not correct. It seems that the `selectItems` function (called by `<nuxeo-results>`) behaves differently in `<nuxeo-data-table>`, causing it to reset the selection instead of adding to the currently selected items.

Reply:
I think we can live with this.

Reply:
[Added a reference to this discussion on ELEMENTS-1297.](https://jira.nuxeo.com/browse/ELEMENTS-1297?focusedCommentId=534789&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-534789)